infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -133,37 +141,54 @@
      * Initialization task.
      * This task is blocked until the management server is bootstrapped and has a connected runtime.
      * Performs recovery if required.
-     * Bootstraps the primary sequencer if the server has been bootstrapped.
-     * Initiates the failure detection and healing detection tasks.
+     * Initiates the monitoring services which performs failure detection and healing detection tasks.
      */
     private void initializationTask() {
         log.info("Start initialization task");
 
+        // Wait for management server bootstrapped.
         try {
             while (!shutdown && serverContext.getManagementLayout() == null) {
-                log.warn("Management Server waiting to be bootstrapped");
+                log.warn("initializationTask: Management Server waiting to be bootstrapped");
                 Sleep.MILLISECONDS.sleepRecoverably(CHECK_BOOTSTRAP_INTERVAL.toMillis());
             }
+        } catch (InterruptedException e) {
+            // Due to shutdown, which interrupts this thread
+            log.debug("initializationTask: Initialization task interrupted without " +
+                    "management server bootstrapped");
+            return;
```

Comment:
Should we throw UnrcoverableCorfuException ?

Reply:
It's ok to return here because this interrupt is caused by shutdown()

Reply:
ok, if you are sure, I agree